internal/lvmd/lvservice.go

```diff
@@ -245,8 +204,14 @@ func (s *lvService) CreateLVSnapshot(ctx context.Context, req *proto.CreateLVSna
 		logger.Error(err, "failed to get thinpool")
 		return nil, status.Error(codes.Internal, err.Error())
 	}
-	tpu, err := pool.Free(ctx)
-	free := calcThinPoolFreeBytes(dc.ThinPoolConfig.OverprovisionRatio, tpu.SizeBytes, tpu.VirtualBytes)
+	poolUsage, err := pool.Usage(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get pool usage: %v", err)
+	}
+	free, err := poolUsage.Free(dc.ThinPoolConfig.OverprovisionRatio)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get free bytes: %v", err)
+	}
```

Comment:
Why don't you use `storagePoolForDeviceClass` and `storagePool` here?

Reply:
The current code modification simply aims to eliminate duplicated processing. However, now that we have `storagePool` interface, I’m wondering if we could push all the concrete thin-pool and VG logic behind it. Right now, some of that implementation is hidden while some remains exposed, and that’s where I’m getting stuck.

For this PR, though, let’s just focus on cleaning up the duplicate code.